source/slang/slang-ir-specialize.cpp

```diff
@@ -770,17 +770,91 @@ struct SpecializationContext
         }
     }
 
+    // Finds any `IRTargetDecoration` from `inst`. Recursively chasing `specialize` chains.
+    IRTargetIntrinsicDecoration* findTargetIntrinsicDecorationRec(IRInst* inst)
```

Comment:
This is duplicated with `CLikeSourceEmitter::findTargetIntrinsicDecoration`, also there is a global scope `findTargetInstrinsicDecoration(IRInst* inst, String targetName)` method but it does not look through `specialize` and `generic` insts. Let me know if you would like to clean this up to have a single `findTargetIntrinsicDecoration` method.

Reply:
I think we should aspire to do  the cleanup, but it should probably be its own PR just to make sure that we can test/validate just the refactoring in isolation.